### PR TITLE
Update `monai.bundle` to support model zoo NGC hosting

### LINKED ISSUE #6761: Update `monai.bundle` to support model zoo NGC hosting
Related changes in model zoo:
https://github.com/Project-MONAI/model-zoo/pull/463

I will add more information later

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #6761 .

### Description

- update `download` to support monaihosting download.
- update relevant unit tests.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by KumoLiu** [comment]
In this PR, I mainly update the `download` method.
It appears that there are still several other methods that need to be updated, such as `get_all_bundles_list`, `get_bundle_versions`, and `get_bundle_info`. 

The[ NGC API doc](https://docs.ngc.nvidia.com/api/) does not, however, contain the API for the current monaihosting link. I can't currently get access to all of the bundles that exist; I can only get access to all of the version information given the name of a specific bundle. Do you think I can refer to the [model_info.json](https://github.com/Project-MONAI/model-zoo/blob/dev/models/model_info.json) method to update these methods or do you have any other suggestions?
cc @wyli @ericspod @Nic-Ma @yiheng-wang-nv

File: tests/test_bundle_download.py [hunk 1]
```diff
@@ -58,6 +58,12 @@
     "model.ts",
 ]
 
+TEST_CASE_6 = [
+    ["models/model.pt", "models/model.ts", "configs/train.json"],
+    "brats_mri_segmentation",
```
Comment (KumoLiu) [inline tests/test_bundle_download.py:63]: Is it necessary to upload a fake model to monaihosting because the real one here may download more slowly?

Comment (wyli) [reply tests/test_bundle_download.py:63]: may include a minimal example from https://github.com/ericspod/tutorials/blob/bundles_update/bundle/bundle_intro.ipynb once that's completed

**Comment by yiheng-wang-nv** [comment]
cc @zephyrie , model zoo web page may also need to be changed due to updated NGC hosting

**Pull request merged**